lib/sycamore/sycamore/tests/unit/inmempyarrowfs.py

```diff
@@ -0,0 +1,98 @@
+import io
+
+from pyarrow.fs import FileSystem, FileSelector, FileInfo, FileType
+
+
+class InMemPyArrowFileSystem(FileSystem):
+    def __init__(self):
+        self._fs = {}
+
+    def copy_file(self, src, dest):
+        assert False, "unimplemented"
+
+    def create_dir(self, path, *, recursive=True):
+        assert isinstance(path, str)
+        # We're blob-like, so create dir is noop.
+        pass
+
+    def delete_dir(self, path):
+        assert False, "unimplemented"
+
+    def delete_dir_contents(self, path, missing_dir_ok=False):
+        assert isinstance(path, str)
+        assert missing_dir_ok, "unimplemented"
+        path = path + "/"
+        todelete = []
+        for k, v in self._fs.items():
+            if k.startswith(path):
+                todelete.append(k)
+
+        for k in todelete:
+            del self._fs[k]
+
+    def delete_file(self, path):
+        assert isinstance(path, str)
+        assert path in self._fs
+        del self._fs[path]
+
+    def equals(self, other):
+        assert False, "unimplemented"
+
+    def get_file_info(self, p):
+        if isinstance(p, str):
+            if p not in self._fs:
+                return FileInfo(str(p))
+
+            # API docs claim we can leave mtime & size as None
+            return FileInfo(str(p), type=FileType.File)
+
+        assert isinstance(p, FileSelector)
+        assert p.allow_not_found, "unimplemented"
+        assert not p.recursive, "unimplemented"
+        dir = p.base_dir + "/"
+        dlen = len(dir)
+        ret = []
+        for k, v in self._fs.items():
+            if not k.startswith(dir):
+                continue
+            if "/" in k[dlen:]:
+                continue
+            ret.append(FileInfo(str(k), type=FileType.File))
+
+        return ret
+
+    def move(self, src, dest):
+        assert False, "unimplemented"
```

Comment:
there's also `raise NotImplementedError()`. Not sure what you prefer.

Reply:
switched. I have no strong preference.